src/components/region-common.js

```diff
@@ -17,8 +17,48 @@ export const regionInfo = (location, locationType) => {
       }
     case 'region':
       switch (location) {
+        case 'ASIA-EAST1':
+          return { flag: '🇹🇼', regionDescription: `${locationType}: ${location} (Taiwan)` }
+        case 'ASIA-EAST2':
+          return { flag: '🇭🇰', regionDescription: `${locationType}: ${location} (Hong Kong)` }
+        case 'ASIA-NORTHEAST1':
+          return { flag: '🇯🇵', regionDescription: `${locationType}: ${location} (Tokyo)` }
+        case 'ASIA-NORTHEAST2':
+          return { flag: '🇯🇵', regionDescription: `${locationType}: ${location} (Osaka)` }
+        case 'ASIA-NORTHEAST3':
+          return { flag: '🇰🇷', regionDescription: `${locationType}: ${location} (Seoul)` }
+        case 'ASIA-SOUTH1':
+          return { flag: '🇮🇳', regionDescription: `${locationType}: ${location} (Mumbai)` }
+        case 'ASIA-SOUTHEAST1':
+          return { flag: '🇸🇬', regionDescription: `${locationType}: ${location} (Singapore)` }
+        case 'ASIA-SOUTHEAST2':
+          return { flag: '🇮🇩', regionDescription: `${locationType}: ${location} (Jakarta)` }
+        case 'AUSTRALIA-SOUTHEAST1':
+          return { flag: '🇦🇺', regionDescription: `${locationType}: ${location} (Sydney)` }
         case 'EUROPE-NORTH1':
           return { flag: '🇫🇮', regionDescription: `${locationType}: ${location} (Finland)` }
+        case 'EUROPE-WEST1':
+          return { flag: '🇧🇪', regionDescription: `${locationType}: ${location} (Belgium)` }
+        case 'EUROPE-WEST2':
+          return { flag: '🇬🇧', regionDescription: `${locationType}: ${location} (London)` }
+        case 'EUROPE-WEST3':
+          return { flag: '🇩🇪', regionDescription: `${locationType}: ${location} (Frankfurt)` }
+        case 'EUROPE-WEST4':
+          return { flag: '🇳🇱', regionDescription: `${locationType}: ${location} (Netherlands)` }
+        case 'EUROPE-WEST6':
+          return { flag: '🇨🇭', regionDescription: `${locationType}: ${location} (Zurich)` }
+        case 'NORTHAMERICA-NORTHEAST1':
+          return { flag: '🇨🇦', regionDescription: `${locationType}: ${location} (Montreal)` }
+        case 'SOUTHAMERICA-EAST1':
+          return { flag: '🇧🇷', regionDescription: `${locationType}: ${location} (Sao Paulo)` }
+        case 'US-CENTRAL1':
+        case 'US-EAST1':
+        case 'US-EAST4':
+        case 'US-WEST1':
+        case 'US-WEST2':
+        case 'US-WEST3':
+        case 'US-WEST4':
+          return { flag: '🇺🇸', regionDescription: `${locationType}: ${location}` }
```

Comment:
Should this show a city/country/whatever like the others do?

Reply:
Looked at this a little bit more, and I think this is fine as is. For non-US regions its sometimes important to know the city/country for regulatory reasons. Its less important for US regions, and showing the region name (e.g. US-WEST4) should suffice.

Reply:
Actually now that I've worked on https://github.com/DataBiosphere/terra-ui/pull/2420/ I think we need the US cities too. I've added that.